homeassistant/config.py

```diff
@@ -473,6 +480,40 @@ def _format_config_error(ex: vol.Invalid, domain: str, config: Dict) -> str:
     return message
 
 
+def _set_time_zone(hass: HomeAssistant, time_zone_str: Optional[str]) -> None:
+    """Help to set the time zone."""
+    if time_zone_str is None:
+        return
+
+    time_zone = date_util.get_time_zone(time_zone_str)
+
+    if time_zone:
+        hass.config.time_zone = time_zone
+        date_util.set_default_time_zone(time_zone)
+    else:
+        _LOGGER.error("Received invalid time zone %s", time_zone_str)
+
+
+async def async_load_ha_core_config(hass: HomeAssistant) -> None:
+    """Store [homeassistant] core config."""
+    store = hass.helpers.storage.Store(CORE_STORAGE_VERSION, CORE_STORAGE_KEY,
+                                       private=True)
+    data = await store.async_load()
+    if data:
+        hac = hass.config
+        hac.config_source = SOURCE_STORAGE
+        hac.latitude = data['latitude']
```

Comment:
How will this handle users backrolling? If we decide to change the schema here, some of these keys might be invalid. If a user then tries backporting to a version before the schema change this will raise an exception - and since this is core code it could (I think) stop HA from starting up.

Reply:
OK, how about validating the data with a schema, and rejecting it if it doesn't pass?

Reply:
Yeah that could probably work